src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -944,18 +964,68 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
       case Success(outputs) =>
         logger.info(s"Collection complete for Scattered Call ${collector.tag}.")
-        self ! CallCompleted(collector, outputs, 0)
+        self ! CallCompleted(collector, outputs, 0, "")
     }
 
     Success(ExecutionStartResult(Set(StartEntry(collector, ExecutionStatus.Starting))))
   }
 
+  private def sendStartMessage(callKey: CallKey, callInputs: Map[String, WdlValue]) = {
+    def registerAbortFunction(abortFunction: AbortFunction): Unit = {}
+    val backendCall = backend.bindCall(workflow, callKey, callInputs, AbortRegistrationFunction(registerAbortFunction))
+
+    def loadCachedBackendCallAndMessage(descriptor: WorkflowDescriptor, cachedExecution: Execution) = {
+      descriptor.namespace.resolve(cachedExecution.callFqn) match {
+        case Some(c: Call) =>
+          val cachedCall = backend.bindCall(
+            descriptor,
+            CallKey(c, cachedExecution.index.toIndex),
+            callInputs,
+            AbortRegistrationFunction(registerAbortFunction)
+          )
+          logger.info(s"Call Caching: Cache hit. Using UUID(${cachedCall.workflowDescriptor.shortId}):${cachedCall.call.name} as results for UUID(${backendCall.workflowDescriptor.shortId}):${backendCall.call.name}")
+          self ! UseCachedCall(callKey, CallActor.UseCachedCall(cachedCall, backendCall))
+        case _ =>
+          logger.error(s"Call Caching: error when resolving '${cachedExecution.callFqn}' in workflow with execution ID ${cachedExecution.workflowExecutionId}: falling back to normal execution")
+          self ! InitialStartCall(callKey, CallActor.Start)
+      }
+    }
+
+    /* Tries to use the cached Execution to send a UseCachedCall message.  If anything fails, send an InitialStartCall message */
+    def loadCachedCallOrInitiateCall(cachedDescriptor: Try[WorkflowDescriptor], cachedExecution: Execution) = cachedDescriptor match {
+        case Success(descriptor) => loadCachedBackendCallAndMessage(descriptor, cachedExecution)
+        case Failure(ex) =>
+          logger.error(s"Call Caching: error when loading workflow with execution ID ${cachedExecution.workflowExecutionId}: falling back to normal execution", ex)
+          self ! InitialStartCall(callKey, CallActor.Start)
+    }
+
+    val hash = backendCall.hash
+    globalDataAccess.getExecutionsWithResuableResultsByHash(hash) onComplete {
+      case Success(executions) if executions.nonEmpty =>
+        val cachedExecution = executions.head
+        globalDataAccess.getWorkflow(cachedExecution.workflowExecutionId) onComplete { cachedDescriptor =>
+          loadCachedCallOrInitiateCall(cachedDescriptor, cachedExecution)
+        }
+      case Success(_) =>
+        self ! InitialStartCall(callKey, CallActor.Start)
+      case Failure(ex) =>
+        logger.error(s"Call Caching: Failed to look up executions that matched hash '$hash'. Falling back to normal execution", ex)
+        self ! InitialStartCall(callKey, CallActor.Start)
+    }
```

Comment:
I feel like this could be refactored to a slicker for comp somehow but it might be a pain to do and maybe not worth it.